perun-core/src/test/java/cz/metacentrum/perun/core/impl/modules/attributes/urn_perun_user_attribute_def_virt_tcsMailsTest.java

```diff
@@ -0,0 +1,195 @@
+	Attribute o365MailsAttr = setUpUserAttribute(5, "o365EmailAddresses:mu", ArrayList.class.getName(), new ArrayList(Arrays.asList(email1, email3, email5)));
+
+	private PerunSessionImpl sess;
+
+	@Before
+	public void setUp() throws Exception {
+		tcsMailsAttrDef.setId(100);
+		//prepare mocks
+		sess = mock(PerunSessionImpl.class);
+		PerunBl perunBl = mock(PerunBl.class);
+		AttributesManagerBl am = mock(AttributesManagerBl.class);
+		UsersManagerBl um = mock(UsersManagerBl.class);
+		ModulesUtilsBl mu = mock(ModulesUtilsBl.class);
+		when(sess.getPerunBl()).thenReturn(perunBl);
+		when(perunBl.getAttributesManagerBl()).thenReturn(am);
+		when(perunBl.getUsersManagerBl()).thenReturn(um);
+		when(perunBl.getModulesUtilsBl()).thenReturn(mu);
+		when(mu.getUserFromMessage(eq(sess), any(String.class))).thenReturn(user);
+		when(sess.getPerunBl().getAttributesManagerBl().getAttribute(sess, user, preferredMailAttr.getName())).thenReturn(preferredMailAttr);
+		when(sess.getPerunBl().getAttributesManagerBl().getAttribute(sess, user, isMailAttr.getName())).thenReturn(isMailAttr);
+		when(sess.getPerunBl().getAttributesManagerBl().getAttribute(sess, user, publicMailsAttr.getName())).thenReturn(publicMailsAttr);
+		when(sess.getPerunBl().getAttributesManagerBl().getAttribute(sess, user, privateMailsAttr.getName())).thenReturn(privateMailsAttr);
+		when(sess.getPerunBl().getAttributesManagerBl().getAttribute(sess, user, o365MailsAttr.getName())).thenReturn(o365MailsAttr);
+		when(sess.getPerunBl().getAttributesManagerBl().getAttributeDefinition(sess, tcsMailsAttrDef.getName())).thenReturn(tcsMailsAttrDef);
+	}
+
+	private Attribute setUpUserAttribute(int id, String friendlyName, String type, Object value) {
+		AttributeDefinition attrDef = new AttributeDefinition();
+		attrDef.setId(id);
+		attrDef.setFriendlyName(friendlyName);
+		attrDef.setNamespace(AttributesManager.NS_USER_ATTR_DEF);
+		attrDef.setType(type);
+		Attribute attr = new Attribute(attrDef);
+		attr.setValue(value);
+		return attr;
+	}
+
+	@Test
+	public void getAttributeValue() throws Exception {
+		@SuppressWarnings("unchecked") ArrayList<String> attributeValue = classInstance.getAttributeValue(sess, user, tcsMailsAttrDef).valueAsList();
+		assertThat(attributeValue, is(notNullValue()));
+		//we want to be sure, that preferredEmail is first (defined by sorting in module)
+		assertTrue(attributeValue.get(0).equals(email1));
+		assertThat(attributeValue, CoreMatchers.hasItem(email1));
+		assertThat(attributeValue, CoreMatchers.hasItem(email2));
+		assertThat(attributeValue, CoreMatchers.hasItem(email3));
+		assertThat(attributeValue, CoreMatchers.hasItem(email4));
+		assertThat(attributeValue, CoreMatchers.hasItem(email5));
+		assertThat(attributeValue.size(), is(5));
+	}
+
+	@Test
+	public void resolveVirtualAttributeValueChangeSet1() throws Exception {
+		AttributeDefinition countries = classInstance.getAttributeDefinition();
```

Comment:
You have this unused variable ``countries`` in all tests. Why?

Reply:
Copy & paste. Removed.